### PR TITLE
Auth lambda

### PR DESCRIPTION
**Pull request by mcallegari10**
## Summary

- Created a Now Lambda for Github Authentication middleware under the `api` folder
- Add Github Login Button component
- Keep some spinners components as examples when not logged in, also added a note to log in
- Change the component card style but it's a WIP (new designs are welcome)

## Screenshot

image

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  `https://github.com/login/oauth/access_token?client_id=${process.env.CLIENT_ID}&client_secret=${process.env.CLIENT_SECRET}&code=${code}`;
+
+module.exports = (req, res) => {
+  https.get(url(req.query.code), response => {
+    let data = '';
+    response.on('data', chunk => { data += chunk; });
+
+    response.on('end', () => { res.send(data); });
+  }).on('error', err => {
+    res.status(500).send(err.message);
```
Comment (FrankIglesias) [inline api/index.js]: Define something different as an error code or forward the whole error object

File: src/services/loginService.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const getTokenFromResponse = queryParams => queryParams.match(/access_token=[a-z0-9]+/g)[0].substring('access_token='.length)
+
+export const storeToken = response => {
+  const token = getTokenFromResponse(response.data)
+  localStorage.setItem('auth_token', token)
+  api.defaults.headers['Authorization'] = `bearer ${token}`
+}
+
+export const loginToGithub = code => axios.get(`https://cookbook-api.wolox-fea.now.sh?code=${code}`).then(response => {
```
Comment (FrankIglesias) [inline src/services/loginService.js]: define the base URL in the .env file

File: src/resources/spinners.js [hunk 3]
```diff
@@ -28,10 +28,10 @@ export default [
...
...ius:50%;content:' ';height:6px;margin:-3px 0 0 -3px;position:absolute;width:6px}.spinner-3 div:nth-child(1){animation-delay:-.036s}.spinner-3 div:nth-child(1)::after{left:50px}.spinner-3 div:nth-child(2){animation-delay:-.072s}.spinner-3 div:nth-child(2)::after{left:45px}.spinner-3 div:nth-child(3){animation-delay:-.108s}.spinner-3 div:nth-child(3)::after{left:40px}.spinner-3 div:nth-child(4){animation-delay:-.144s}.spinner-3 div:nth-child(4)::after{left:35px}.spinner-3 div:nth-child(5){animation-delay:-.18s}.spinner-3 div:nth-child(5)::after{left:30px}.spinner-3 div:nth-child(6){animation-delay:-.216s}.spinner-3 div:nth-child(6)::after{left:25px}.spinner-3 div:nth-child(7){animation-delay:-.252s}.spinner-3 div:nth-child(7)::after{left:20px}.spinner-3 div:nth-child(8){animation-delay:-.288s}.spinner-3 div:nth-child(8)::after{left:15px}.spinner-3 div:nth-child(1)::after{top:50px}.spinner-3 div:nth-child(2)::after{top:54px}.spinner-3 div:nth-child(3)::after{top:57px}.spinner-3 div:nth-child(4)::after{top:58px}.spinner-3 div:nth-child(5)::after{top:57px}.spinner-3 div:nth-child(6)::after{top:54px}.spinner-3 div:nth-child(7)::after{top:50px}.spinner-3 div:nth-child(8)::after{top:45px}`
```
Comment (SKOLZ) [inline src/resources/spinners.js]: wtf

Comment (mcallegari10) [reply src/resources/spinners.js]: Mocked components to show some examples when not logged in 😄

**Pull request merged**